graphql/schemabuilder/types.go

```diff
@@ -331,6 +336,10 @@ type method struct {
 	// FederationType is an object where all the fields are keys
 	// that can be exposed over federation
 	FederationType interface{}
+
+	// ShadowObjectType is an object where all the fields are keys
+	// that can be fetched in a federated subquery
+	ShadowObjectType interface{}
```

Comment:
Maybe change the comment to say 
"ShadowObjectType is the reflect type of parent object if it is a shadow object" ?

Reply:
Also, do you think we can change the type of this field and the field above to reflect.Type? why we need it to be an interface here?